docs/proposals/resource-aggregation-proxy/README.md

```diff
@@ -0,0 +1,455 @@
+    - member2
+    - member3
+  resourceSelectors:
+    - apiVersion: v1
+      kind: Pod
+    - apiVersion: v1
+      kind: Node
+```
+
+
+
+## How to Merge Result
+
+- Get request: return first result.
+- List request: concat the results, and return to client as a single list.
+- Watch request: return result one by one.
+
+## Resource Version
+
+Kubernetes uses ResourceVersion to uniquely identify every version of a resource. We merge all clusters' ResourceVersion into one JSON struct, `base64.RawURLEncoding` it, and get the final string.
+
+```
+base64.RawURLEncoding({"cluster1":"1234","cluster2":"5678"}) = "eyJjbHVzdGVyMSI6IjEyMzQiLCJjbHVzdGVyMiI6IjU2NzgifQ"
+```
+
+## Example
+
+There are two member clusters: cluster1 and cluster2.
+
+|                 | cluster1         | cluster2         |
+| --------------- | ---------------- | ---------------- |
+| resourceVersion | 1234             | 5678             |
+| pods            | nginx-1, nginx-2 | nginx-3, nginx-4 |
+
+**List Pod**
+
+Situation: Client wants to list all pods in cluster1 and cluster2.
+
+
+
+1\. Client sends request with prefixed url.
+
+```
+GET /apis/proxy.karmada.io/v1alpha1/globalresources/dummy/proxy/api/v1/namespaces/default/pods
+```
+
+2\. Proxy query clusters1's cache, get nginx-1, nginx-2; query cluster2's cache, get nginx-3, nginx-4.
+
+3\. Proxy merges pods' list and returns.
+
+```yaml
+apiVersion: v1
+kind: List
+metadata:
+  # eyJjbHVzdGVyMSI6IjEyMzQiLCJjbHVzdGVyMiI6IjU2NzgifQ is result of 
+  # base64.RawURLEncoding("{"cluster1":"1234","cluster2":"5678"}")
+  resourceVersion: eyJjbHVzdGVyMSI6IjEyMzQiLCJjbHVzdGVyMiI6IjU2NzgifQ
+items:
+- apiVersion: v1
+  kind: Pod
+  metadata:
+    name: nginx-1
+  ...
+- apiVersion: v1
+  kind: Pod
+  metadata:
+    name: nginx-2
+  ...
+- apiVersion: v1
+  kind: Pod
+  metadata:
+    name: nginx-3
+  ...
+- apiVersion: v1
+  kind: Pod
+  metadata:
+    name: nginx-4
+  ...
+```
+
+
+
+**Get Pod**
+
+Situation: Client wants to get Pod by name "nginx-2".
+
+
+
+1\. Client sends request with prefixed url.
+
+```
+GET /apis/proxy.karmada.io/v1alpha1/globalresources/dummy/proxy/api/v1/namespaces/default/pods/nginx-2
+```
+
+2\. Proxy queries all caches, and finds out the required Pod is in cluster1's cache.
+
+3\. Proxy modifies the resourceVersion field then returns response to client.
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nginx-2
+  namespace: default
+  resourceVersion: eyJjbHVzdGVyMSI6IjEyMzQiLCJjbHVzdGVyMiI6IjU2NzgifQ
+  ...
+```
+
+
+
+**Watch Pod**
+
+Situation: Client wants to watch Pod.
+
+
+
+1\. Client sends request with prefixed url.
+
+```
+GET /apis/proxy.karmada.io/v1alpha1/globalresources/dummy/proxy/api/v1/namespaces/default/pods?resourceVersion=eyJjbHVzdGVyMSI6IjEyMzQiLCJjbHVzdGVyMiI6IjU2NzgifQ&watch=true
+```
+
+2\. Proxy watches cluster1 and cluster2's cache.
+
+3\. When there is change, proxy merges the event, modifies the resourceVersion field, then returns response to client.
```

Comment:
1. In the watch situation. The proposal will collect the member clusters events and re-allocate global resouceVersion ？
2. Accutally we now use clusterpedia to do `Get` operation for collect member clusters and use  aa to router clusterpedia and karmada.  And we plan to impl watch by `1`.  I think it is same with your proposal. 
Maybe we can help to join your proposal.

Reply:
> re-allocate global resouceVersion ？

you mean the RV is independent from member RV? In fact, we do not want to maintain this RV. The global RV is built with member RVs in a json. See it in `Resource Version` section.

> 2\. Maybe we can help to join your proposal.

Thanks~